File: javascript/src/api_internal/endpoint.ts
// this file is @generated

import {
  type EndpointTransformationIn,
  EndpointTransformationInSerializer,
} from "../models/endpointTransformationIn";
import { EndpointAutoConfigDeprecated } from "./endpointAutoConfigDeprecated";
import { EndpointAutoconfig } from "./endpointAutoconfig";
import { HttpMethod, SvixRequest, type SvixRequestContext } from "../request";

export class Endpoint {
  public constructor(private readonly requestCtx: SvixRequestContext) {}

  public get autoConfigDeprecated() {
    return new EndpointAutoConfigDeprecated(this.requestCtx);
  }

  public get autoconfig() {
    return new EndpointAutoconfig(this.requestCtx);
  }

  /**
   * This operation was renamed to `set-transformation`.
   *
   * @deprecated
   */
  public async transformationPartialUpdate(
    appId: string,
    endpointId: string,
    endpointTransformationIn: EndpointTransformationIn = {}
  ): Promise<void> {
    const request = new SvixRequest(
      HttpMethod.PATCH,
      "/api/v1/app/{app_id}/endpoint/{endpoint_id}/transformation"
    );

    request.setPathParam("app_id", appId);
    request.setPathParam("endpoint_id", endpointId);
    request.setBody(
      EndpointTransformationInSerializer._toJsonObject(endpointTransformationIn)
    );

    return await request.sendNoResponseBody(this.requestCtx);
  }
}

File: rust/src/internal/api/endpoint.rs
// this file is @generated
use super::EndpointAutoConfig;
use crate::{error::Result, models::*, Configuration};

pub struct Endpoint<'a> {
    cfg: &'a Configuration,
}

impl<'a> Endpoint<'a> {
    pub(super) fn new(cfg: &'a Configuration) -> Self {
        Self { cfg }
    }

    pub fn auto_config(&self) -> EndpointAutoConfig<'a> {
        EndpointAutoConfig::new(self.cfg)
    }

    /// This operation was renamed to `set-transformation`.
    #[deprecated]
    pub async fn transformation_partial_update(
        &self,
        app_id: String,
        endpoint_id: String,
        endpoint_transformation_in: EndpointTransformationIn,
    ) -> Result<()> {
        crate::request::Request::new(
            http1::Method::PATCH,
            "/api/v1/app/{app_id}/endpoint/{endpoint_id}/transformation",
        )
        .with_path_param("app_id", app_id)
        .with_path_param("endpoint_id", endpoint_id)
        .with_body_param(endpoint_transformation_in)
        .returns_nothing()
        .execute(self.cfg)
        .await
    }
}
